core/templates/domain/skill/concept-card-backend-api.service.ts

```diff
@@ -17,92 +17,99 @@
  * about the concept card of a skill from the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require('domain/skill/skill-domain.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
 
-angular.module('oppia').factory('ConceptCardBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService', 'CONCEPT_CARD_DATA_URL_TEMPLATE',
-  function($http, $q, UrlInterpolationService, CONCEPT_CARD_DATA_URL_TEMPLATE) {
-    // Maps previously loaded concept cards to their IDs.
-    var _conceptCardCache = [];
+import cloneDeep from 'lodash/cloneDeep';
 
-    var _fetchConceptCards = function(
-        skillIds, successCallback, errorCallback) {
-      var conceptCardDataUrl = UrlInterpolationService.interpolateUrl(
-        CONCEPT_CARD_DATA_URL_TEMPLATE, {
-          comma_separated_skill_ids: skillIds.join(',')
-        });
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { SkillDomainConstants } from
+  'domain/skill/skill-domain.constants';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ConceptCardBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolation: UrlInterpolationService) {}
 
-      $http.get(conceptCardDataUrl).then(function(response) {
-        var conceptCards = angular.copy(response.data.concept_card_dicts);
+  // Maps previously loaded concept cards to their IDs.
+  private _conceptCardCache = [];
+
+  private _fetchConceptCards(
+      skillIds: Array<string>, successCallback: any, errorCallback: any): any {
+    var conceptCardDataUrl = this.urlInterpolation.interpolateUrl(
+      SkillDomainConstants.CONCEPT_CARD_DATA_URL_TEMPLATE, {
+        comma_separated_skill_ids: skillIds.join(',')
+      });
+
+    this.http.get(conceptCardDataUrl).toPromise().then(
+      (response: any) => {
         if (successCallback) {
-          successCallback(conceptCards);
+          successCallback(response.concept_card_dicts);
         }
-      }, function(errorResponse) {
+      }, (errorResponse) => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(errorResponse.body);
```

Comment:
I think this should be `errorResponse.error`. Could you check?

Reply:
I saw various different usages in merged PRs (`errorResponse`, `errorResponse.body`, or `errorResponse.error`) but just changed this line to `errorResponse.error`!

Reply:
Yes, I agree different usages have been used but let's stick to errorResponse.error.